src/aya/lifecycle.md

```diff
@@ -0,0 +1,68 @@
+# Program Lifecycle
+
+## BPF Object Lifecycle
+
+BPF object lifecycle is tied to File Descriptors (FDs) returned from the `bpf` syscall.
+Every time you create a program or map, retrieve a reference to a program/map, or attach
+a program, you are given an FD and the Kernel increments its internal reference counter.
+When the FD is closed, the reference count is decreased.
+When it reaches 0, the kernel will unload/detach the BPF object.
+
+There are some expections to the above, for example, when attachments are handled using another API (e.g netlink)
+
+If you wish for your BPF object to outlive the program that creates it, you may
+pin a BBPF object to BPFFS `/sys/fs/bpf`.
+
+## Aya Object Lifecycle
+
+When you call `load` or `load_file`, the FDs returned by map and program creation are
```

Comment:
I'd say something like:

In Aya, an instance of the `Bpf` type manages the lifetime of all the eBPF objects created through it.

When you call [load](...) or [load_file](...), all the maps referenced by the eBPF code are created and stored inside the returned `Bpf` instance:

```
example code with comments
```

When you load a program included in a `Bpf` instance, the program stays attached until the parent `Bpf` instance gets dropped:

```
example with comments
```

When you attach a program, the program stays attached until the parent `Bpf` instance is dropped, or until you call `detach()` on the returned [LinkRef](link to linkref) value:

```
example with comments
```